src/Libplanet.Types/Assets/Currency.cs

```diff
@@ -418,27 +128,71 @@ private Currency(SerializationInfo info, StreamingContext context)
         private Currency(
             string ticker,
             byte decimalPlaces,
+            (BigInteger Major, BigInteger Minor)? maximumSupply,
             IImmutableSet<Address>? minters,
             bool totalSupplyTrackable)
         {
-            ticker = ticker.Trim();
-
-            if (string.IsNullOrEmpty(ticker))
-            {
-                throw new ArgumentException(
-                    "Currency ticker symbol cannot be empty.",
-                    nameof(ticker)
-                );
-            }
-
+            ValidateArguments(ticker, decimalPlaces, maximumSupply, minters, totalSupplyTrackable);
             Ticker = ticker;
             Minters = minters;
             DecimalPlaces = decimalPlaces;
-            _maximumSupply = null;
+            _maximumSupply = maximumSupply;
             TotalSupplyTrackable = totalSupplyTrackable;
-            Hash = GetHash(Minters, Ticker, DecimalPlaces, _maximumSupply, TotalSupplyTrackable);
         }
 
+        private Currency(Dictionary dictionary)
+            : this(
+                ExtractTicker(dictionary),
+                ExtractDecimalPlaces(dictionary),
+                ExtractMaximumSupply(dictionary),
+                ExtractMinters(dictionary),
+                ExtractTotalSupplyTrackable(dictionary))
+        {
+        }
+
+        private Currency(SerializationInfo info, StreamingContext context)
+            : this(_codec.Decode((byte[])info.GetValue("ByteEncoded", typeof(byte[]))!))
```

Comment:
Shouldn't there be code to read the old format as well? I know it's going away soon, but